product_sale_manufactured_for/models/res_partner.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get("active") is False:
+            query = (
+                "DELETE FROM product_product_manuf_for_partner_rel "
+                "WHERE partner_id IN (%s);"
+            )
+            self.env.cr.execute(query, (tuple(self.ids)))
```

Comment:
Maybe, as the information of whom the product was manufactured for is lost when a customer is archived. This could be logged in the chatter ?

Reply:
Good idea.
Regarding the override of write, wouldn't be less "intrusive" to define and inverse method for the active field only?